contracts/PetPark.sol

```diff
@@ -4,4 +4,93 @@ pragma solidity ^0.8.0;
 
 contract PetPark {
 
+    address  contractOwner;
+    enum AnimalType {
+        NONE, FISH, CAT, DOG, RABBIT, PARROT
+    }
+    enum Gender{
+        MALE, FEMALE
+    }
+    struct Animal {
+        AnimalType animalType;
+        uint count;
+    }
+
+    struct Borrower {
+        Gender gender;
+        uint age;
+        uint petCount;
+        bool exists;
+        AnimalType animalType;
+    }
+
+    event Added(AnimalType animalType, uint animalCount);
+    event Borrowed(AnimalType animalType);
+    event Returned(AnimalType animalType);
+
+    mapping(AnimalType => Animal) animals;
+    mapping(address => Borrower)borrowers;
+
+    constructor(){
+        contractOwner = msg.sender;
+    }
+
+    /*
+    * @dev method to add animals to the contract.@author Satish
+    */
+    function add(AnimalType _animalType, uint _count) external onlyOwner {
+        require(_animalType != AnimalType.NONE, "Invalid animal");
+        animals[_animalType].animalType = _animalType;
```

Comment:
Since `animals[_animalType]` is being accessed multiple times better to save it in a separate variable